tripy/examples/resent50/resnet50.ipynb

```diff
@@ -0,0 +1,529 @@
+    "\n",
+    "    def __call__(self, x):\n",
+    "        for stage in self.stages:\n",
+    "            x = stage(x)\n",
+    "        return x"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "class ResNetModel(tp.Module):\n",
+    "    def __init__(self):\n",
+    "        super(ResNetModel, self).__init__()\n",
+    "        self.embedder = ResNetEmbeddings()\n",
+    "        layers_config = [\n",
+    "            (3, 256, 64, (1, 1)),\n",
+    "            (4, 512, 128, (2, 2)),\n",
+    "            (6, 1024, 256, (2, 2)),\n",
+    "            (3, 2048, 512, (2, 2)),\n",
+    "        ]\n",
+    "        self.encoder = ResNetEncoder(layers_config)\n",
+    "\n",
+    "    def __call__(self, x):\n",
+    "        x = self.embedder(x)\n",
+    "        x = self.encoder(x)\n",
+    "        x = tp.avgpool(x, kernel_dims=(7, 7), stride=(7, 7)) # output size will be (1, 1)\n",
+    "        return x"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "class ResNetClassifier(tp.Module):\n",
+    "    def __init__(self, num_classes=1000):\n",
+    "        super(ResNetClassifier, self).__init__()\n",
+    "        self.resnet = ResNetModel()\n",
+    "        self.classifier = tp.Linear(2048, num_classes)\n",
+    "\n",
+    "    def __call__(self, x):\n",
+    "        features = self.resnet(x)\n",
+    "        features = tp.flatten(features, start_dim=1)\n",
+    "        output = self.classifier(features)\n",
+    "        return output"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Instantiate the Tripy model\n",
+    "tripy_model = ResNetClassifier()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Load weights to Tripy Model\n",
+    "\n",
+    "First we ensure all state dict keys match and then we load the pretrained weights."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def load_pretrained_weights_tp(model, pretrained_state_dict):\n",
+    "    \"\"\"\n",
+    "    This function loads weights from a state_dict into the custom TPResNet model,\n",
+    "    converting them into tp.Parameter before loading.\n",
+    "    \"\"\"\n",
+    "    model_dict = model.state_dict()  # Get the model's state dict\n",
+    "\n",
+    "    # Convert each weight to a tp.Parameter and update the model's state dict\n",
+    "    for k, v in pretrained_state_dict.items():\n",
+    "        if 'num_batches_tracked' in k:\n",
+    "            continue    # Skip num_batches_tracked since it is not needed for inference (tp.BatchNorm does not support)\n",
+    "        if 'classifier' in k:\n",
+    "            k = k.replace('1.', '') # Remove extra 1 (slight naming mismtach)\n",
```

Comment:
Any idea why this mismatch is happening?

Reply:
because torch does sequential on `Flatten` and `Linear` for the classification head. We have sequential but can't mix them until the mixed container PR is in.

Reply:
Makes sense. Can we do the mixed container PR first then so we can fix this before merging?

Reply:
Merged mixed container, and fixed this now!